docs/index.rst

```diff
@@ -315,3 +315,43 @@ Configuration Options
   * Type: long
   * Default: 60000
   * Importance: low
+
+
+Schema Evolution
+----------------
+
+The JDBC connector supports schema evolution. When there is a change in a database table schema,
+the JDBC connector can detect the change, create a new Kafka connect schema and try to register
+a new Avro schema in the schema registry. Whether we can successfully register the schema or not
+depends on the compatibility level of the schema registry which is backward by default.
+
+For example, if we remove a column from the table, the change is a backward compatible and the
+corresponding Avro schema can be successfully registered in the schema registry. If we modify
+the database table schema to change a column type or add a column, when the Avro schema is
+registered to the schema registry, it will be rejected as the changes are not backward compatible.
+
+You can change the compatibility level of schema registry to allow non compatible schema or other
+compatibility levels. There are two ways to do this:
+
+* Configure the schema registry to use other schema compatibility level by setting
+  ``avro.compatibility.level`` in schema registry. Note that this is a global setting that applies
+  to all schemas in the schema registry.
+
+* Set the compatibility level for subjects which are used by the connector using
+  ``PUT /config/(string: subject)``. The subjects have format of ``topic-key`` and ``topic-value``
+  where the ``topic`` is determined by ``topic.prefix`` config and table name.
+
+However, due to the limitation of the JDBC API, some compatible schema changes may be treated as
+non compatible change. For example, adding a column with default value is a backward compatible
+change. However, there is no JDBC API to get the default value for a column. Even the change of
```

Comment:
This isn't strictly true -- there is COLUMN_DEF in the database metadata. We might need to phrase this differently. (I think there are limitations, e.g. it is the string form of the default, not sure what you'd do with defaults that are a function such as CURRENT_TIMESTAMP, etc.)